src/applications/vre/definitions/profileAddress.js

```diff
@@ -0,0 +1,265 @@
+ */
+
+// Corresponding working form schema:
+export const addressSchema = {
+  type: 'object',
+  properties: {
+    isMilitary: {
+      type: 'boolean',
+    },
+    'view:livesOnMilitaryBaseInfo': {
+      type: 'object',
+      properties: {},
+    },
+    country: {
+      type: 'string',
+      enum: countries.map(country => country.value),
+      enumNames: countries.map(country => country.label),
+    },
+    addressLine1: {
+      type: 'string',
+      minLength: 1,
+      maxLength: 100,
+      pattern: '^.*\\S.*',
+    },
+    addressLine2: {
+      type: 'string',
+      minLength: 1,
+      maxLength: 100,
+      pattern: '^.*\\S.*',
+    },
+    addressLine3: {
+      type: 'string',
+      minLength: 1,
+      maxLength: 100,
+      pattern: '^.*\\S.*',
+    },
+    city: {
+      type: 'string',
+    },
+    state: {
+      type: 'string',
+    },
+    postalCode: {
+      type: 'string',
+    },
+  },
+};
+const MILITARY_STATES = Object.entries(ADDRESS_DATA.states).reduce(
+  (militaryStates, [stateCode, stateName]) => {
+    if (ADDRESS_DATA.militaryStates.includes(stateCode)) {
+      return {
+        ...militaryStates,
+        [stateCode]: stateName,
+      };
+    }
+    return militaryStates;
+  },
+  {},
+);
+
+const USA = {
+  value: 'USA',
+  label: 'United States',
+};
+
+const MilitaryBaseInfo = () => (
+  <div className="vads-u-padding-x--2p5">
+    <AdditionalInfo
+      status="info"
+      triggerText="Learn more about military base addresses"
+    >
+      <span>
+        The United States is automatically chosen as your country if you live on
+        a military base outside of the country.
+      </span>
+    </AdditionalInfo>
+  </div>
+);
+
+/**
+ * @param {string} path - path to the address in formData, may contain [INDEX] as part of it, which needs to be handled using insertArrayIndex
+ * @param {string} checkBoxTitle - Visual label for the military base checkbox. Ex: "I live on a United States military base outside of the U.S."
+ * @param {function} callback - slots into ui:required for the necessary fields
+ *
+ * Conventions:
+ * 1. formDataPath - path to entire address property in formData, after accounting for potential array nesting. Derived from the path parameter
+ * 2. get(formDataPath, formData) - returns the address property, often destructured with const {country, isMilitary} = get(formDataPath, formData)
+ *
+ * Examples:
+ * 1. Path to Address nested in array - childrenToAdd[INDEX].childAddressInfo.address
+ */
+
+export const addressUiSchema = (path, checkBoxTitle, callback) => {
+  // Used when addresses are nested in an array and need to be assessible.
+  const insertArrayIndex = (key, index) => key.replace('[INDEX]', `[${index}]`);
+  // the address is nested in an array, handle index
+  const getPath = (pathToData, index) =>
+    typeof index === 'number'
+      ? insertArrayIndex(pathToData, index)
+      : pathToData;
+
+  return (function returnAddressUI() {
+    return {
```

Comment:
Maybe I'm confused... doesn't an IIFE that returns an object, do the same thing as just returning the object? Is it necessary?

Reply:
You're right, it isn't necessary 👍 I think I thought it might have been, in order to keep some things in scope, but now that you mention it, it works just fine without the IIFE. 

_time to go back and re-read documentation on IIFEs_